enhancements/sig-architecture/33-hub-templates-securing-sensitive-data/README.md

```diff
@@ -0,0 +1,268 @@
+is backed up during backup and restore situations. With this in place, the
+Policy Propagator is able to encrypt the data using the `protect` template
+function, however, the managed cluster also needs the key when it decrypts the
+data. The Policy Spec Sync controller should also sync the
+`policy-encryption-key` secret to the managed cluster in the managed cluster
+namespace by watching the managed cluster namespace for secrets on the Hub. It
+is also desired to allow customers to provide their own keys. This would be
+limited to AES-256 keys unless a customer requests otherwise.
+
+One thing to consider is if we want to add periodic automatic key rotations per
+managed cluster. This would add security, however, it would cause all policies
+to be reprocessed by the Policy Propagator. All policies that used the `protect`
+template function would need to be replicated to the managed clusters. This
+could cause a lot of churn in large scale scenarios. Additionally, there would
+be an odd period where the key is changed on the Hub and the managed cluster,
+but the recomputed policies with the new key would not be replicated yet, thus
+the Configuration Policy controller would fail to decrypt the values and mark
+all policies as non-compliant. There are some workarounds for this, but it is
+proposed to skip this for the upcoming release and provide documentation to
+manually rotate the key in the event a customer detects that it has been
+compromised. This will likely be made automatic in a future release.
+
+#### RBAC
+
+Since the encryption key is stored in a secret in the managed cluster namespace,
+it is important that non-privileged users do not have access to read this value.
+The default ACM roles of `open-cluster-management:view:<managed_cluster_name>`
+and `open-cluster-management:admin:<managed_cluster_name>` do not provide such
+access and thus the managed cluster namespace on the Hub is a secure location
+for the secret to reside. That being said, the admin role should be granted
+access to rotate the key. Depending on the implementation, this may be as simple
+as granting access to delete the secret. This still needs to be properly
+documented so that users' custom RBAC configuration protects the secret.
+
+### Risks and Mitigation
+
+See the Implementation Details section.
+
+## Design Details
+
+See the Implementation Details section.
+
+### Test Plan
+
+Test cases:
+
+1. Test that the `protect` hub template function encrypts the data on the
+   replicated policy and it can be decrypted on the managed cluster
+1. Test that the `fromSecret` hub template function encrypts the data on the
+   replicated policy and it can be decrypted on the managed cluster.
+
+### Graduation Criteria
+
+It is proposed that this be considered stable in the first release after the
+implementation is complete.
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
```

Comment:
I think version skew should be discussed, For instance, what if you have a high version hub control plane and a lower version agent?